spec/services/notifications/moderation/send_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "rails_helper"
+
+RSpec.describe Notifications::Moderation::Send do
+  let(:last_moderation_time) { Time.zone.now - User.moderator_availability_delay - 3.hours }
+  let(:dev_account) { create(:user) }
+  let(:user) { create(:user) }
+  let(:article) { create(:article, user_id: user.id) }
+  let(:comment) { create(:comment, user: user, commentable: article) }
+  let(:available_moderators) { User.with_role(:trusted).where("last_moderation_notification < ?", User.moderator_availability_delay.ago) }
+
+  before do
+    create(:user, :trusted, last_moderation_notification: last_moderation_time)
+    allow(User).to receive(:dev_account).and_return(dev_account)
+    # Creating a comment calls moderation job which itself call moderation service
+    Comment.skip_callback(:create, :after, :send_to_moderator)
+  end
+
+  after do
+    Comment.set_callback(:create, :after, :send_to_moderator)
+  end
+
+  it "calls comment_data since parameter is a comment" do
+    allow(Notifications).to receive(:comment_data)
+    described_class.call(available_moderators, comment)
+    expect(Notifications).to have_received(:comment_data)
+  end
+
+  it "checks wether created Notification is valid", :aggregate_failuresi do
+    expect do
+      described_class.call(available_moderators, comment)
+    end.to change(Notification, :count).by(1)
+
+    last = Notification.last
```

Comment:
I suggest putting expectations about the notification instance in a separate spec.
And I suspect that relying on the last notification is not 100% safe (records could hypothetically be created in the database while the test is run). I'm in favor of returning a notifications instance from the service and checking the returned object attributes.

Reply:
I added the return of the notification instance, and updated the tests. Easier to test, I was not sure the first time whether I could change the code.